### PR TITLE
Alpine based docker image.

### PR DESCRIPTION
**Pull request by FireMasterK**
This PR reduces the size of the docker image from 1.09GB to 468MB.

### CODE REVIEW

**Comment by FireMasterK** [comment]
I'd ideally like to have a production environment in docker too, perhaps with docker-compose files.

**Comment by FireMasterK** [comment]
I've also changed it to use gunicorn instead of flask, is that ok?

**Comment by pluja** [comment]
Thats perfect! Make the pull request when it's ready and I will merge it.

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,21 @@
-FROM python:3
+FROM python:alpine
```
Comment (SuperSandro2000) [inline Dockerfile:1]:
```
FROM python:3-alpine
```

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
 
 WORKDIR /usr/src/app
 
+RUN apk add gcc musl-dev libffi-dev openssl-dev libxml2-dev libxslt-dev file llvm-dev make g++
```
Comment (SuperSandro2000) [inline Dockerfile:5]: Those dev dependencies should ideally be removed in the same layer again.

File: Dockerfile [hunk 3]
```diff
@@ -1,15 +1,21 @@
...
+
 COPY requirements.txt ./
+
+RUN pip install wheel cryptography gunicorn pymysql
```
Comment (SuperSandro2000) [inline Dockerfile:9]: ``--no-cache-dir``

Why are those deps not in the requirements.txt?

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+version: '3.4'
+services:
+  mariadb:
+    image: mariadb:latest
```
Comment (SuperSandro2000) [inline docker-compose.yml:4]: Pint that to the latest version cause major updates definitely break the DB.

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    environment:
+      MYSQL_ROOT_PASSWORD: changeme
+      MYSQL_DATABASE: yotter
+      MYSQL_USER: yotter
+      MYSQL_PASSWORD: changeme
+    restart: on-failure
+    volumes:
+     - /var/lib/mysql
+    healthcheck:
```
Comment (SuperSandro2000) [inline docker-compose.yml:13]: Remove that. It just delays the time until the container is ready in a single container setup.

**Comment by FireMasterK** [comment]
Thanks for the feedback! I'll fix these soon

File: docker-compose.yml (hunk 5 above)
Comment (FireMasterK) [reply docker-compose.yml:13]: Remove what? The restart policy or the health check?

File: Dockerfile (hunk 3 above)
Comment (FireMasterK) [reply Dockerfile:9]: Those dependencies can be found in [here](https://github.com/pluja/Yotter/blob/dev-indep/SELF-HOSTING.md#manual-installation) looks like they're only used for production environments.

**Comment by FireMasterK** [comment]
Checkout #40 I have implemented some changes in accordance over there.

**Pull request merged**